sklearn/mixture/bayesian_mixture.py

```diff
@@ -0,0 +1,650 @@
+                                           ensure_2d=False)
+            _check_shape(self.mean_prior_, (n_features, ), 'means')
+
+    def _check_precision_parameters(self, X):
+        """Check the prior parameters of the precision distribution.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+        """
+        _, n_features = X.shape
+
+        if self.nu_prior is None:
+            self.nu_prior_ = n_features
+        elif self.nu_prior > n_features - 1.:
+            self.nu_prior_ = self.nu_prior
+        else:
+            raise ValueError("The parameter 'nu_prior' "
+                             "should be greater than %d, but got %.3f."
+                             % (n_features - 1, self.nu_prior))
+
+    def _checkcovariance_prior__parameter(self, X):
+        """Check the `covariance_prior_`.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+        """
+        _, n_features = X.shape
+
+        if self.covariance_prior is None:
+            self.covariance_prior_ = {
+                'full': np.atleast_2d(np.cov(X.T)),
+                'tied': np.atleast_2d(np.cov(X.T)),
+                'diag': np.var(X, axis=0, ddof=1),
+                'spherical': np.var(X, axis=0, ddof=1).mean()
+            }[self.covariance_type]
+
+        elif self.covariance_type in ['full', 'tied']:
+            self.covariance_prior_ = check_array(
+                self.covariance_prior, dtype=[np.float64, np.float32],
+                ensure_2d=False)
+            _check_shape(self.covariance_prior_, (n_features, n_features),
+                         '%s covariance_prior' % self.covariance_type)
+            _check_precision_matrix(self.covariance_prior_,
+                                    self.covariance_type)
+        elif self.covariance_type is 'diag':
+            self.covariance_prior_ = check_array(
+                self.covariance_prior, dtype=[np.float64, np.float32],
+                ensure_2d=False)
+            _check_shape(self.covariance_prior_, (n_features,),
+                         '%s covariance_prior' % self.covariance_type)
+            _check_precision_positivity(self.covariance_prior_,
+                                        self.covariance_type)
+        # spherical case
+        elif self.covariance_prior > 0.:
+            self.covariance_prior_ = self.covariance_prior
+        else:
+            raise ValueError("The parameter 'spherical covariance_prior' "
+                             "should be greater than 0., but got %.3f."
+                             % self.covariance_prior)
+
+    def _initialize(self, X, resp):
+        """Initialization of the mixture parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+
+        resp : array-like, shape (n_samples, n_components)
+        """
+        nk, xk, sk = _estimate_gaussian_parameters(X, resp, self.reg_covar,
```

Comment:
A lot of things are already been called means and covariances on the algorithm. 
It will create a lot of mistakes for people who want to analyse the code. 
Moreover, nk, xk and sk are the notation used by the original book and the documentation I will add later. 
As the equations are difficult to understand and implement, I think it's better to keep the original notation.